### PR TITLE
feat: Migrate Contributions Table

### LINKED ISSUE #2129: 🚀 [Feature] Migrate Contribution Table for Deny
## 🚀 Feature
Add columns to contributions table:
 * `denied_at  datetime DEFAULT NULL AFTER confirmed_at`
 * `denied_by int(10) unsigned DEFAULT NULL AFTER denied_at`
 * `contribution_type varchar(12) COLLATE utf8mb4_unicode_ci NOT NULL DEFAULT ADMIN`
 *  `contribution_status varchar(12) COLLATE utf8mb4_unicode_ci NOT NULL DEFAULT PENDING`

Enumerator for contribution type:
 * `ADMIN` if `moderator_id` is present
 * `USER` if both, `moderator_id` and `contribution_link_id` are NOT present
 * `LINK` if `contribution_link_id` is present

Ensue that the `contribution_type` is set correctly for old and new contributions.

Enumerator for contribution status:
 * `PENDING`
 * `DELETED`
 * `IN_PROGRESS`
 * `CONFIRMED`
 * `DENIED`
 
Contributions with `deleted_at` not null are `DELETED`
Contributions with `confitmed_at` not null are `CONFIRMED`
All other contributions are `PENDING`

### PR DESCRIPTION
**Pull request by Mogge**
## 🍰 Pullrequest
Add columns contribution type, contribution status, denied at and denied by to contributions table. Add enumerators for contribution type and contribution status. Set type and status accordingly.

### Issues
- fixes #2129 

### Todo

- [X] None

### CODE REVIEW

File: backend/src/graphql/enum/ContributionStatus.ts [hunk 1]
```diff
@@ -0,0 +1,14 @@
+import { registerEnumType } from 'type-graphql'
+
+export enum ContributionStatus {
+  PENDING = 'PENDING',
+  DELETED = 'DELETED',
+  IN_PROGRESS = 'IN_PROGRESS',
```
Comment (Elweyn) [inline backend/src/graphql/enum/ContributionStatus.ts:6]: What the difference with PENDING? IN_PROGRESS is when their has been a question back?

**Review by Elweyn — APPROVED** [review]
Looks prety good! Tested the migration up and down, and the USER and ADMIN Creation working as well!

**Review by Tirokk — CHANGES_REQUESTED** [review]
Functions very good.

I tired type link, admin, and user and it all works fine. 🚀💫💫

Pleace have a look at my notices of contributions definition !!!
But I have a question …

File: backend/src/graphql/enum/ContributionStatus.ts (hunk 1 above)
Comment (Mogge) [reply backend/src/graphql/enum/ContributionStatus.ts:6]: PENDING: A moderator has to check the contribution.
IN_PROGRESS: Moderator has questions (added a message) and is waiting for user response
Please resolve conversation when your question is answered completely

File: database/entity/0045-add_denied_type_and_status_to_contributions/Contribution.ts [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+  moderatorId: number
+
+  @Column({ unsigned: true, nullable: true, name: 'contribution_link_id' })
+  contributionLinkId: number
+
+  @Column({ unsigned: true, nullable: true, name: 'confirmed_by' })
+  confirmedBy: number
+
+  @Column({ nullable: true, name: 'confirmed_at' })
+  confirmedAt: Date
```
[the remark this answers is not in the document]
Comment (Mogge) [reply database/entity/0045-add_denied_type_and_status_to_contributions/Contribution.ts:54]: The reason may be the `@DeleteDateColumn` decorator does not allow `nullable: true`, but I am not sure. As long as typescript does not run in errors I am fine. The mechanism works as it is.

File: database/entity/0045-add_denied_type_and_status_to_contributions/Contribution.ts [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+
+  @Column({ unsigned: true, nullable: true, name: 'denied_by' })
+  deniedBy: number
+
+  @Column({ nullable: true, name: 'denied_at' })
+  deniedAt: Date
```
[the remark this answers is not in the document]
Comment (Mogge) [reply database/entity/0045-add_denied_type_and_status_to_contributions/Contribution.ts:60]: Same

File: database/entity/0045-add_denied_type_and_status_to_contributions/Contribution.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (Tirokk) [reply database/entity/0045-add_denied_type_and_status_to_contributions/Contribution.ts:54]: Okidoki

File: database/entity/0045-add_denied_type_and_status_to_contributions/Contribution.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (Tirokk) [reply database/entity/0045-add_denied_type_and_status_to_contributions/Contribution.ts:60]: Okidoki

**Review by Tirokk — APPROVED** [review]
Okidoki.

I approve.

**Pull request merged**